flink-python/pyflink/datastream/connectors/kinesis.py

```diff
@@ -0,0 +1,368 @@
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+from typing import Dict, Union, List
+
+from pyflink.common import SerializationSchema, DeserializationSchema
+from pyflink.datastream.functions import SourceFunction
+from pyflink.datastream.connectors import Sink
+from pyflink.java_gateway import get_gateway
+
+
+# ---- KinesisSource ----
+
+
+class KinesisShardAssigner(object):
+    """
+    Utility to map Kinesis shards to Flink subtask indices. Users can implement this interface to
+    optimize distribution of shards over subtasks. See assign(StreamShardHandle, int) for details.
+    """
+
+    def __init__(self, _j_kinesis_shard_assigner):
+        self._j_kinesis_shard_assigner = _j_kinesis_shard_assigner
+
+    @staticmethod
+    def uniform_shard_assigner() -> 'KinesisShardAssigner':
+        return get_gateway().jvm.org.apache.flink.streaming.connectors \
+            .kinesis.util.UniformShardAssigner()
+
+
+class KinesisDeserializationSchema(object):
+    """
+    This is a deserialization schema specific for the Flink Kinesis Consumer. Different from the
+    basic DeserializationSchema, this schema offers additional Kinesis-specific information about
+    the record that may be useful to the user application.
+    """
+
+    def __init__(self, _j_kinesis_deserialization_schema):
+        if _j_kinesis_deserialization_schema is None:
+            self._j_kinesis_deserialization_schema = get_gateway().jvm.org.apache.flink \
+                .streaming.connectors.kinesis.serialization.KinesisDeserializationSchema
+        else:
+            self._j_kinesis_deserialization_schema = _j_kinesis_deserialization_schema
+
+
+class AssignerWithPeriodicWatermarks(object):
+    """
+    The AssignerWithPeriodicWatermarks assigns event time timestamps to elements, and generates
+    low watermarks that signal event time progress within the stream. These timestamps and
+    watermarks are used by functions and operators that operate on event time, for example event
+    time windows.
+    """
+
+    def __init__(self, _j_assigner_with_periodic_watermarks):
+        self._j_assigner_with_periodic_watermarks = _j_assigner_with_periodic_watermarks
+
+
+class WatermarkTracker(object):
+    """
+    The watermark tracker is responsible for aggregating watermarks across distributed operators.
+    It can be used for sub tasks of a single Flink source as well as multiple heterogeneous sources
+    or other operators.The class essentially functions like a distributed hash table that enclosing
+    operators can use to adopt their processing / IO rates
+    """
```

Comment:
Add static method job_manager_watermark_tracker which returns JobManagerWatermarkTracker? Otherwise, however users use it?